### PR TITLE
Add issue and PR templates 

### LINKED ISSUE #8350: Implement Issue and PR templates
As suggested by @brechmos-stsci , it might be beneficial to give users a check list when they open a new issue or a new pull request (the latter should be consistent with existing dev guide). A first draft could be obtained from existing sources, for example https://www.talater.com/open-source-templates .

For instance, choosing new issue, bug + feature request, and backend brought me to https://www.talater.com/open-source-templates/#/page/98

```
<!--- Provide a general summary of the issue in the Title above -->

## Expected Behavior
<!--- If you're describing a bug, tell us what should happen -->
<!--- If you're suggesting a change/improvement, tell us how it should work -->

## Current Behavior
<!--- If describing a bug, tell us what happens instead of the expected behavior -->
<!--- If suggesting a change/improvement, explain the difference from current behavior -->

## Possible Solution
<!--- Not obligatory, but suggest a fix/reason for the bug, -->
<!--- or ideas how to implement the addition or change -->

## Steps to Reproduce (for bugs)
<!--- Provide a link to a live example, or an unambiguous set of steps to -->
<!--- reproduce this bug. Include code to reproduce, if relevant -->
1.
2.
3.
4.

## Context
<!--- How has this issue affected you? What are you trying to accomplish? -->
<!--- Providing context helps us come up with a solution that is most useful in the real world -->

## Your Environment
<!--- Include as many relevant details about the environment you experienced the bug in -->
* Version used:
* Environment name and version (e.g. PHP 5.4 on nginx 1.9.1):
* Server type and version:
* Operating System and version:
* Link to your project:
```

### PR DESCRIPTION
**Pull request by pllim**
Fix #8350

Template blatantly copied from SunPy. Thanks, @Cadair !

### CODE REVIEW

**Comment by pllim** [comment]
p.s. Need a repo admin to activate them, I think. That is beyond what I can do here.

File: .github/ISSUE_TEMPLATE/bug_report.md [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+3. [and so on...]
+
+```python
+# Put your Python code snippet here.
+```
+
+### System Details
+<!-- Even if you do not think this is necessary, it is still nice to have. -->
+
+<!-- import astropy; print(astropy.__version__) -->
```
Comment (bsipocz) [inline .github/ISSUE_TEMPLATE/bug_report.md]: I like at sklearn land that they have a copy paste-able block for this:
```
import platform; print(platform.platform())
import sys; print("Python", sys.version)
import numpy; print("NumPy", numpy.__version__)
import astropy; print("Astropy", astropy.__version__)
```

**Comment by bsipocz** [comment]
Do we want to point out the minimal failing example page at SO? https://stackoverflow.com/help/minimal-reproducible-example

(Also at sklearn, they say more about how an idea bug report looks like (they also have some  of that detailed out in their template): 
https://scikit-learn.org/dev/developers/contributing.html#how-to-make-a-good-bug-report
)


But either way, I would vote get this merge now than go into bike shedding the details, that can be done in follow-ups after seeing how people interact with the templates.

File: .github/ISSUE_TEMPLATE/bug_report.md (hunk 1 above)
Comment (pllim) [reply .github/ISSUE_TEMPLATE/bug_report.md]: Done.

File: .github/ISSUE_TEMPLATE/bug_report.md [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+1. [First Step]
+2. [Second Step]
+3. [and so on...]
+
+```python
+# Put your Python code snippet here.
+```
+
+### System Details
+<!-- Even if you do not think this is necessary, it is still nice to have.
```
Comment (bsipocz) [inline .github/ISSUE_TEMPLATE/bug_report.md]:
```suggestion
<!-- Even if you do not think this is necessary, it is useful information for the maintainers.
```

**Comment by pllim** [comment]
(Ops... I edited straight on GitHub and forgot to skip ci...)

I agree that additional details can be a future PR. I am afraid if we make the template too long, people won't read it.

**Comment by bsipocz** [comment]
> I agree that additional details can be a future PR. I am afraid if we make the template too long, people won't read it.

yes, agreed. We can adjust if non copy pasteable snippets keep coming.

**Comment by pllim** [comment]
I am going to YOLO a bit here and merge this now, so repo admin can set them up. I think it is important to get this up and running before Hacktoberfest starts.

**Pull request merged**